test/model_simulation/simulate_SDEs.jl

```diff
@@ -93,81 +103,197 @@ let
-            (i == 2) && (u0[1] += 1000.0)
-            (i == 3) ? (p[2:2:6] .*= 1000.0; u0 .+= 1000) : (p[1] += 500.0)
-            prob1 = SDEProblem(networks[1], u0, (0.0, 100.0), p)
-            prob2 = SDEProblem(networks[2][1], networks[2][2], u0, (0.0, 100.0), p,
-                               noise_rate_prototype = networks[2][3])
-            du1 = similar(u0)
-            du2 = similar(u0)
-            prob1.f.f(du1, u0, p, 0.0)
-            prob2.f.f(du2, u0, p, 0.0)
-            @test all(isapprox.(du1, du2))
-            g1 = zeros(numspecies(networks[1]), numreactions(networks[1]))
-            g2 = copy(g1)
-            prob1.f.g(g1, u0, p, 0.0)
-            prob2.f.g(g2, u0, p, 0.0)
-            @test all(isapprox.(g1, g2))
+    push!(catalyst_networks, reaction_networks_constraint[9])
+    push!(manual_networks, (f = real_f_3, g = real_g_3, nrp = zeros(7, 6)))
+    push!(u0_syms, [:X1, :X2, :X3, :X4, :X5, :X6, :X7])
+    push!(ps_syms, [:k1, :k2, :k3, :k4, :k5, :k6])
+
+    for (rn_catalyst, rn_manual, u0_sym, ps_sym) in zip(catalyst_networks, manual_networks, u0_syms, ps_syms)
+        for factor in [1e-1, 1e0], repeat in 1:3
+            # Set input values.
+            u0_1 = rnd_u0(rn_catalyst, rng; factor, min = 100.0)
+            ps_1 = rnd_ps(rn_catalyst, rng; factor, min = 0.01)
+            u0_2 = map_to_vec(u0_1, u0_sym)
+            ps_2 = map_to_vec(ps_1, ps_sym)
+
+            # Check drift functions.
+            dt = zeros(length(u0_1))
+            rn_manual.f(dt, u0_2, ps_2, 0.0)
+            @test dt ≈ f_eval(rn_catalyst, u0_1, ps_1, 0.0)
+
+            # Check diffusion functions.
+            duW = zeros(size(rn_manual.nrp))
+            rn_manual.g(duW, u0_2, ps_2, 0.0)
+            @test duW ≈ g_eval(rn_catalyst, u0_1, ps_1, 0.0)
+
+            # Compares simulation with identical seed. 
+            # Cannot test the third network (requires very fiddly parameters for CLE to be defined).
+            if nameof(rn_catalyst) != :rnc9
+                sprob_1 = SDEProblem(rn_catalyst, u0_1, (0.0, 1.0), ps_1)
+                sprob_2 = SDEProblem(rn_manual.f, rn_manual.g, u0_2, (0.0, 1.0), ps_2; noise_rate_prototype = rn_manual.nrp)
+                sol1 = solve(sprob_1, ImplicitEM(); seed = 1234)
+                sol2 = solve(sprob_2, ImplicitEM(); seed = 1234)
+                @test sol1[u0_sym] ≈ sol2.u
+            end
         end
     end
 end
 
+# Checks that simulations for a large number of potential systems are completed (and don't error).
+# (cannot solve as solution likely to go into negatives). 
+let
+    for rn in reaction_networks_all
+        u0 = rnd_u0(rn, rng)
+        ps = rnd_ps(rn, rng)
+        sprob = SDEProblem(rn, u0, (0.0, 1.0), ps)
+        init(sprob, ImplicitEM())
+    end
+end
+
 ### Noise Scaling ###
 
-# Tests with a single noise scaling parameter.
+# Compares noise scaling with manually declared noise function. 
```

Comment:
This one noise scaling test was added here and not in the noise scaling PR. Because the required helper functions were created in this PR, this test did not come until this one.